code/modules/mob/living/silicon/robot/robot_modules.dm

```diff
@@ -340,6 +340,19 @@
 	model_traits = list(TRAIT_PUSHIMMUNE)
 	hat_offset = 3
 
+/obj/item/robot_model/medical/be_transformed_to(obj/item/robot_model/old_model)
+	var/mob/living/silicon/robot/cyborg = loc
+	var/list/medical_icons = list(
+		"Qualified Doctor" = image(icon = 'icons/mob/robots.dmi', icon_state = "qualified_doctor"),
+		)
```

Comment:
this should really be a static list.

Reply:
this system should be refactored instead of copypasting it for every borg that has special types

Reply:
^^